tests/tibanna/ffcommon/test_generic_qc_utils.py

```diff
@@ -144,4 +177,292 @@ def test_filter_workflow_args_by_property(wf_arg_set_1):
     qc_zip = filter_workflow_args_by_property(qcs, "qc_zipped", True)
     assert len(qc_zip) == 1
     assert qc_zip[0]["workflow_argument_name"] == "qc_input_zip_1"
-    
\ No newline at end of file
+
+
+@pytest.fixture
+def valid_ruleset_1a():
+    return {
+            "qc_thresholds": [
+                {
+                    "id": "c1",
+                    "metric": "Coverage",
+                    "operator": ">=",
+                    "pass_target": 100,
+                    "warn_target": 80,
+                    "use_as_qc_flag": True,
+                },
+                {
+                    "id": "rl",
+                    "metric": "read_length",
+                    "operator": ">",
+                    "pass_target": 120,
+                    "warn_target": 80,
+                    "use_as_qc_flag": True,
+                },
+                {
+                    "id": "ts",
+                    "metric": "Total Sequences [Samtools]",
+                    "operator": ">",
+                    "pass_target": 547993296,
+                    "warn_target": 527993296,
+                },
+                {
+                    "id": "test0",
+                    "metric": "test0",
+                    "operator": "<=",
+                    "pass_target": 10,
+                    "warn_target": 10,
+                },
+                {
+                    "id": "test1",
+                    "metric": "test0",
+                    "operator": "<",
+                    "pass_target": 10,
+                    "warn_target": 10,
+                },
+                {
+                    "id": "test2",
+                    "metric": "test0",
+                    "operator": "==",
+                    "pass_target": 10,
+                    "warn_target": 10,
+                },
+                {
+                    "id": "test3",
+                    "metric": "test0",
+                    "operator": "<>",  # invalid
+                    "pass_target": 10,
+                    "warn_target": 10,
+                },
+            ],
+            "overall_quality_status_rule": "{c1} and {ts}"
+        }
+
+
+@pytest.fixture
+def valid_ruleset_1b():
+    return {
+            "qc_thresholds": [
+                {
+                    "id": "c1",
+                    "metric": "coverage",
+                    "operator": ">=",
+                    "pass_target": 150,
+                    "warn_target": 80,
+                    "use_as_qc_flag": True,
+                },
+            ],
+            "overall_quality_status_rule": "{c1}"
+        }
+
+
+@pytest.fixture
+def valid_ruleset_2a():
```

Comment:
You might consider giving these more meaningful names so they can be deciphered without looking at